hom/cogs/views.py

```diff
@@ -208,33 +252,41 @@ def __init__(self):
         custom_id="persistent_view:message_close",
     )
     async def message_close(
-        self, interaction: discord.Interaction, button: discord.ui.Button
+        self, interaction: discord.Interaction, _: discord.ui.Button["SupportMessage"]
     ) -> None:
+        embed = discord.Embed(description=f"{interaction.user.mention} has closed the ticket.")
+
+        assert isinstance(interaction.channel, discord.guild.GuildChannel)
+        assert isinstance(interaction.user, discord.Member)
```

Comment:
What do these assert statements do if they fail? Not really familiar with the purpose of assert statements so I'm not sure if it's valid but my main concern here would be never responding to the interaction.

Assuming the asserts fail something, the asserts could probably be moved to the top, pushing the embed creation below them.

Reply:
Moved them to the top.